src/pygame_simulator/PySim_noise.py

```diff
@@ -147,155 +213,142 @@ def draw(self, key_points = []):
       
       #self.ball_internal.spin = self.ball_internal.spin - ball_robot_vector/np.linalg.norm(ball_robot_vector) * spin_rot_accel
       self.ball_internal.controler = robot
-      pull_vel = robot.loc + convert_local([robot_radius + ball_radius, 0], robot.rot) - self.ball_internal.loc
-      pull_vel = pull_vel / np.linalg.norm(pull_vel) * max_ball_spin
+      pull_vel = robot.loc + convert_local([self.robot_radius + self.ball_radius, 0], robot.rot) - self.ball_internal.loc
+      pull_vel = pull_vel / np.linalg.norm(pull_vel) * self.max_ball_spin
       #print(pull_vel)
-      self.ball_internal.spin = self.ball_internal.spin * spin_accel_rate + pull_vel * (1-spin_accel_rate)
-      self.ball_internal.velocity = self.ball_internal.velocity * spin_accel_rate + pull_vel * (1-spin_accel_rate)
+      self.ball_internal.spin = self.ball_internal.spin * self.spin_accel_rate + pull_vel * (1-self.spin_accel_rate)
+      self.ball_internal.velocity = self.ball_internal.velocity * self.spin_accel_rate + pull_vel * (1-self.spin_accel_rate)
       #print("spin - ",robot.id, robot.is_blue, " - ", pull_vel, self.ball_internal.velocity)
-      
-    action = robot.run_action()
-    if action == None:
-      action = [0,0,0,0,0]
-      
+    
+  def update_bot_kicker(self, robot, action):
     '''
-    kicking stuff
+    brief: updates a robot's kicker
+    params: robot - robot to update 
+            action - the action taken by the robot
     '''
-      
+    ball_robot_vector = self.ball_internal.loc - robot.loc
+    robot_local_ball_loc = convert_local(ball_robot_vector, -robot.rot)
     if robot.kick_cooldown > 0:
       robot.kick_cooldown -= 1
     if action[0] and robot.kick_cooldown == 0:
       #print("kick")
-      robot.kick_cooldown = KICK_CD
+      robot.kick_cooldown = self.kick_cd
       #print(robot_local_ball_loc[0] > 0, robot_local_ball_loc[0] < robot_radius + kick_length + ball_radius, abs(robot_local_ball_loc[1]) > kick_width/2)
-      if (robot_local_ball_loc[0] > 0 and robot_local_ball_loc[0] < robot_radius + kick_length + ball_radius
-        and abs(robot_local_ball_loc[1]) < kick_width/2):
-        self.ball_internal.velocity = self.ball_internal.velocity + kick_delta_V * ball_robot_vector / np.linalg.norm(ball_robot_vector)
+      if (robot_local_ball_loc[0] > 0 and robot_local_ball_loc[0] < self.robot_radius + self.kick_length + self.ball_radius
+        and abs(robot_local_ball_loc[1]) < self.kick_width/2):
+        self.ball_internal.velocity = self.ball_internal.velocity + self.kick_delta_V * np.array([np.cos(-robot.rot), np.sin(-robot.rot)])
     
+  def update_bot_movement(self, robot, action, delta_time):
     '''
-    movement stuff
+    brief: updates a robot's movement
+    params: robot - robot to update 
+            action - the action taken by the robot
+            delta_time - time since last update
     '''
     
     #15.3846 is there to make control magnitudes similar to GRsim
```

Comment:
Fix action class to have class variables. Please move magic numbers to the class declaration